### DISCUSSION TITLE
How to read/write incrementally

### DISCUSSION THREAD

**Discussion by attermann** [jgromes/RadioLib#902] [Q&A]
I need to send and receive LoRa packets with a 1-byte header. For efficiency I'd like to be able to read/write the 1-byte header to/from a variable first, and then read/write the remaining bytes to/from a separate buffer. The available methods seem geared toward complete packets though. 

For example I tried calling readData first to read 1 byte and then a second time to read the remaining bytes, but that seemed to result in corrupted data. 

Is this possible to do? I suspect it is and I'm just overlooking something simple but haven't been able to find an answer yet in docs or example code.

**Comment by jgromes** [comment]
This is not really supported. SX127x will actively dump data that was not requested by the user when calling `readData`. For other LoRa modules (like SX126x), I think we would need additional logic to maintain consistency between data remaining in buffer and what the user has requested.

Also, I'm not sure it's correct to assume that reading just the header first will be (significantly) more efficient. The library is not just reading the packet buffer, there are other SPI transactions being done (typically querying the packet length checking CRCs, clearing interrupts etc.), which would be done twice in your approach, or we would need to implement additional logic to avoid them. If your application really is limited by having to read the whole packet, then you could try increasing the SPI bus speed, which is quite slow by default (2 MHz).
